downloadModel with additional plugins throws MalformedJsonException on windows

I am trying to run a command line build on windows via gradle `./gradlew.bat downloadModel` and am facing a `com.google.gson.stream.MalformedJsonException`. 

I am using` modelixVersion=2021.1.125`.

My **configuration** looks like this: 

```
modelixModel {
    serverUrl = "https://my-modelix.com/"
    repositoryId = "foo"
    timeout = 18000

    mpsPath = mpsDir
    projectPath = "${rootDir}"
    makeProject = "languages..etc"
    addPlugin("C:\\my-project\\artifacts\\com.mbeddr.platform\\com.mbeddr.mpsutil.actionsfilter", "com.mbeddr.mpsutil.actionsfilter")
}
```

This results in the following **exception**: 

```
Exception in ExportMain
com.google.gson.stream.MalformedJsonException: Unterminated object at line 1 column 10 path $[0].path
com.google.gson.JsonSyntaxException: com.google.gson.stream.MalformedJsonException: Unterminated object at line 1 column 10 path $[0].path
        at com.google.gson.internal.Streams.parse(Streams.java:60)
        at com.google.gson.JsonParser.parseReader(JsonParser.java:85)
        at com.google.gson.JsonParser.parseReader(JsonParser.java:60)
        at com.google.gson.JsonParser.parseString(JsonParser.java:47)
        at org.modelix.gradle.model.EnvironmentLoader.loadAdditionalPlugins(EnvironmentLoader.java:136)
        at org.modelix.gradle.model.EnvironmentLoader.loadEnvironment(EnvironmentLoader.java:220)
        at org.modelix.gradle.model.ExportMain.main(ExportMain.java:53)
Caused by: com.google.gson.stream.MalformedJsonException: Unterminated object at line 1 column 10 path $[0].path
        at com.google.gson.stream.JsonReader.syntaxError(JsonReader.java:1564)
        at com.google.gson.stream.JsonReader.doPeek(JsonReader.java:492)
        at com.google.gson.stream.JsonReader.hasNext(JsonReader.java:415)
        at com.google.gson.internal.bind.TypeAdapters$29.read(TypeAdapters.java:722)
        at com.google.gson.internal.bind.TypeAdapters$29.read(TypeAdapters.java:715)
        at com.google.gson.internal.bind.TypeAdapters$29.read(TypeAdapters.java:698)
        at com.google.gson.internal.Streams.parse(Streams.java:48)
```

How to reproduce: Run `downloadModel` gradle task on windows.

--- a/gradle-plugin/build.gradle
+++ b/gradle-plugin/build.gradle
@@ -18,9 +18,10 @@ repositories {
 
 dependencies {
     compileOnly fileTree(dir: '../artifacts/mps/lib', include: '*.jar')
-    implementation 'com.google.code.gson:gson:2.8.9'
+    implementation 'com.google.code.gson:gson:2.10'
+    implementation 'org.apache.commons:commons-text:1.10.0'
     compile gradleApi()
-    testCompile group: 'junit', name: 'junit', version: '4.12'
+    testCompile group: 'junit', name: 'junit', version: '4.13.1'
 }
 
 compileJava.dependsOn(":mps:resolveMps")

--- a/gradle-plugin/src/main/java/org/modelix/gradle/model/ExportMain.java
+++ b/gradle-plugin/src/main/java/org/modelix/gradle/model/ExportMain.java
@@ -15,6 +15,7 @@ package org.modelix.gradle.model;
 
 import com.intellij.ide.plugins.IdeaPluginDescriptor;
 import com.intellij.ide.plugins.PluginManager;
+import org.apache.commons.text.StringEscapeUtils;
 
 import java.io.File;
 
@@ -25,11 +26,20 @@ public class ExportMain {
     private static boolean DEBUG = false;
 
     private static void processSetting(String key, String value) {
-        System.out.println("(setting) " + key + " = " + value);
-        System.setProperty(key, value);
+        String settingValue = value;
+        if (isJsonValuedProperty(key)) {
+            settingValue = StringEscapeUtils.unescapeJson(settingValue);
+        }
+        System.out.println("(setting) " + key + " = " + settingValue);
+        System.setProperty(key, settingValue);
         if (key.equals(PROPERTY_PREFIX + EnvironmentLoader.Key.DEBUG.getCode())) {
             DEBUG = Boolean.parseBoolean(value);
         }
+    }
+
+    private static boolean isJsonValuedProperty(String key) {
+        return key.endsWith(EnvironmentLoader.Key.ADDITIONAL_PLUGINS.getPropertyKey())
+                || key.endsWith(EnvironmentLoader.Key.ADDITIONAL_PLUGIN_DIRS.getPropertyKey());
     }
 
     private static void debugMessage(String message) {
@@ -41,7 +51,7 @@ public class ExportMain {
     public static void main(String[] args) {
         System.out.println("[ExportMain]");
         try {
-            for (int i = 1; i < args.length; i+=2) {
+            for (int i = 1; i < args.length; i += 2) {
                 String key = PROPERTY_PREFIX + args[i - 1];
                 String value = args[i];
                 processSetting(key, value);

--- a/gradle-plugin/src/main/java/org/modelix/gradle/model/ModelPlugin.java
+++ b/gradle-plugin/src/main/java/org/modelix/gradle/model/ModelPlugin.java
@@ -1,5 +1,6 @@
 package org.modelix.gradle.model;
 
+import org.apache.commons.text.StringEscapeUtils;
 import org.gradle.api.Plugin;
 import org.gradle.api.Project;
 import org.gradle.api.Task;
@@ -119,7 +120,7 @@ public class ModelPlugin implements Plugin<Project> {
             Manifest manifest = readManifest();
             String modelixVersion = manifest.getMainAttributes().getValue("modelix-Version");
             Configuration genConfig = project.getConfigurations().detachedConfiguration(
-                project.getDependencies().create("org.modelix:gradle-plugin:" + modelixVersion)
+                    project.getDependencies().create("org.modelix:gradle-plugin:" + modelixVersion)
             );
 
             final Configuration mpsConfig;
@@ -135,7 +136,7 @@ public class ModelPlugin implements Plugin<Project> {
                 String mpsVersion = manifest.getMainAttributes().getValue("MPS-Version");
                 String extensionsVersion = manifest.getMainAttributes().getValue("MPS-extensions-Version");
                 mpsConfig = project.getConfigurations().detachedConfiguration(
-                        project.getDependencies().create("com.jetbrains:mps:" + mpsVersion )
+                        project.getDependencies().create("com.jetbrains:mps:" + mpsVersion)
                 );
                 pluginsConfig = project.getConfigurations().detachedConfiguration(
                         project.getDependencies().create("de.itemis.mps:extensions:" + extensionsVersion),
@@ -202,8 +203,8 @@ public class ModelPlugin implements Plugin<Project> {
                 );
                 addArgIfNotNullOrBlank(javaExec, Key.ADDITIONAL_LIBRARIES, settings.getAdditionalLibrariesAsString());
                 addArgIfNotNullOrBlank(javaExec, Key.ADDITIONAL_LIBRARY_DIRS, settings.getAdditionalLibraryDirsAsString());
-                addArgIfNotNullOrBlank(javaExec, Key.ADDITIONAL_PLUGINS, settings.getAdditionalPluginsAsString());
-                addArgIfNotNullOrBlank(javaExec, Key.ADDITIONAL_PLUGIN_DIRS, settings.getAdditionalPluginDirsAsString());
+                addArgIfNotNullOrBlank(javaExec, Key.ADDITIONAL_PLUGINS, StringEscapeUtils.escapeJson(settings.getAdditionalPluginsAsString()));
+                addArgIfNotNullOrBlank(javaExec, Key.ADDITIONAL_PLUGIN_DIRS, StringEscapeUtils.escapeJson(settings.getAdditionalPluginDirsAsString()));
                 addArgIfNotNullOrBlank(javaExec, Key.PROJECT, settings.getProjectFile());
                 addArgIfNotNullOrBlank(javaExec, Key.MPS_EXTENSIONS_PATH, settings.getMpsExtensionsArtifactsPath());
                 addArgIfNotNullOrBlank(javaExec, Key.MODELIX_PATH, settings.getModelixArtifactsPath());

--- a/mps/org.modelix.model.mpsplugin/models/org.modelix.model.mpsplugin.mps
+++ b/mps/org.modelix.model.mpsplugin/models/org.modelix.model.mpsplugin.mps
@@ -10868,9 +10868,14 @@
                 <ref role="3cqZAo" node="29etMtbjK8W" resolve="fileSystem" />
               </node>
               <node concept="liA8E" id="29etMtbjLQr" role="2OqNvi">
-                <ref role="37wK5l" to="3ju5:~IFileSystem.getFile(java.lang.String)" resolve="getFile" />
-                <node concept="37vLTw" id="29etMtbjLQs" role="37wK5m">
-                  <ref role="3cqZAo" node="29etMtbjIqk" resolve="exportPath" />
+                <ref role="37wK5l" to="3ju5:~IFileSystem.getFile(java.io.File)" resolve="getFile" />
+                <node concept="2ShNRf" id="7mdB9oJ$3TE" role="37wK5m">
+                  <node concept="1pGfFk" id="7mdB9oJ$9Ou" role="2ShVmc">
+                    <ref role="37wK5l" to="guwi:~File.&lt;init&gt;(java.lang.String)" resolve="File" />
+                    <node concept="37vLTw" id="7mdB9oJ$dRX" role="37wK5m">
+                      <ref role="3cqZAo" node="29etMtbjIqk" resolve="exportPath" />
+                    </node>
+                  </node>
                 </node>
               </node>
             </node>

